core-primitives/stf-executor/src/enclave_signer.rs

```diff
@@ -91,12 +106,28 @@ where
 		shard: &ShardIdentifier,
 	) -> Result<TrustedCallSigned> {
 		let mr_enclave = self.ocall_api.get_mrenclave_of_self()?;
-		let enclave_account_nonce = self.get_enclave_account_nonce(shard)?;
+		let enclave_account_id = self.get_enclave_account()?;
 		let enclave_call_signing_key = self.get_enclave_call_signing_key()?;
 
+		let current_nonce = self.get_enclave_account_nonce(shard)?;
+		let pending_tx = self
+			.top_pool_author
+			.get_pending_trusted_calls(shard.clone())
+			.iter()
+			.filter(|v| match v {
+				TrustedOperation::indirect_call(ref call) =>
+					call.call.sender_account().eq(&enclave_account_id),
+				TrustedOperation::direct_call(ref call) =>
+					call.call.sender_account().eq(&enclave_account_id),
+				_ => false,
+			})
+			.count();
```

Comment:
I can add that, but may I ask who will use it eventually? Or only in tests? 

I'll add a unit test too

Reply:
I would think that if we expose a rpc like that it could be a useful debugging tool, but I have not thought this through. However, I think wrapping this into a dedicated function is good practice anyhow.